File: frontend/src/app/dtos/cau-hoi/cauhoi-dto.ts
export class CauHoiDTO {
  bo_cau_hoi_id!: number;
  noi_dung!: string;
  do_kho: 'DE' | 'TRUNG_BINH' | 'KHO' = 'TRUNG_BINH';
  loai_noi_dung: 'VAN_BAN' | 'HINH_ANH' | 'AM_THANH' | 'VIDEO' = 'VAN_BAN';
  duong_dan_tep?: string | null;
  lua_chon_a!: string;
  lua_chon_b!: string;
  lua_chon_c!: string;
  lua_chon_d!: string;
  dap_an_dung!: 'A' | 'B' | 'C' | 'D';
  giai_thich?: string;

  [key: string]: any;

  constructor(data?: Partial<CauHoiDTO>) {
    if (data) {
      Object.assign(this, data);
    }
  }
}

File: backend/src/main/java/com/app/backend/dtos/CauHoiDTO.java
package com.app.backend.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CauHoiDTO {
    @NotNull
    @JsonProperty("bo_cau_hoi_id")
    private Long boCauHoiId;

    @JsonProperty("do_kho")
    private String doKho; // DE | TRUNG_BINH | KHO

    @NotBlank
    @JsonProperty("noi_dung")
    private String noiDung;

    @JsonProperty("loai_noi_dung")
    private String loaiNoiDung; // VAN_BAN | HINH_ANH | AM_THANH | VIDEO

    @JsonProperty("duong_dan_tep")
    private String duongDanTep;

    @NotBlank
    @JsonProperty("lua_chon_a")
    private String luaChonA;

    @NotBlank
    @JsonProperty("lua_chon_b")
    private String luaChonB;

    @NotBlank
    @JsonProperty("lua_chon_c")
    private String luaChonC;

    @NotBlank
    @JsonProperty("lua_chon_d")
    private String luaChonD;

    @JsonProperty("dap_an_dung")
    private Character dapAnDung; // A|B|C|D

    @JsonProperty("giai_thich")
    private String giaiThich;
}